crates/proof-of-sql/src/sql/parse/query_context_builder.rs

```diff
@@ -162,17 +165,23 @@ impl<'a> QueryContextBuilder<'a> {
     ) -> ConversionResult<ColumnType> {
         let left_dtype = self.visit_expr(left)?;
         let right_dtype = self.visit_expr(right)?;
-        check_dtypes(left_dtype, right_dtype, op)?;
+        check_dtypes(left_dtype, right_dtype, op.clone())?;
```

Comment:
As to @iajoiner's comment:
> P.S. I don't really think we need to make changes in this PR to satisfy `clippy::needless_pass_by_value` since we pass in `UnaryOp` by value as well and clippy doesn't flag that.

The reason `UnaryOp` should be passed by value, but `BinaryOp` should be passed by reference is that `UnaryOp` implements `Copy`, but `BinaryOp` does not.

Reply:
In general, if you can fix a clippy lint, you should. It is almost always a good suggestion.
In this case, I agree with clippy's suggestion, even though it is a bit [pedantic](https://github.com/spaceandtimelabs/sxt-proof-of-sql/blob/531a2ba40c119a534c9ce09295255cfb530bc59c/Cargo.toml#L74).
If anything, we should do it the way clippy suggests so we don't get in the bad habit of just allowing lints when they are inconvenient.

Reply:
@JayWhite2357 Some `BinaryOperator` variants contain strings hence it does make sense for it to not implement `Copy`. That doesn't happen for `UnaryOperator`.